commands/pull_request.go

```diff
@@ -156,6 +159,16 @@ func pullRequest(cmd *Command, args *Args) {
 	fullBase := fmt.Sprintf("%s:%s", baseProject.Owner, base)
 	fullHead := fmt.Sprintf("%s:%s", headProject.Owner, head)
 
+	if web {
+		openPullRequestURL := fmt.Sprintf("https://github.com/%s/compare/%s...%s?expand=1", baseProject, base, fullHead)
+		launcher, err := utils.BrowserLauncher()
+		utils.Check(err)
+		args.Replace(launcher[0], "", launcher[1:]...)
```

Comment:
`args.Replace` sets `args.Executable`, `args.Command` and `args.Params`, in that order.

I assume that when `pullRequest` returns, `args` is executed. Before this line of the code, `args.Executable` may be `git`, and `args.Command` will be `pull-request`. (`Params` get consumed).

This line sets `args.Executable` to `open`, at least on a Mac. It behaves differently for other platforms, like Windows, where it also sets `args.Arguments`: https://github.com/dasilvacontin/hub/blob/open-web-interface/utils/utils.go#L25-L57.

This piece of code is actually copied from some part below in the same file: https://github.com/github/hub/blob/4d40b6ee887b76dfe88ca2debfcfc9685fbc4bb7/commands/pull_request.go#L214-L218.